Sprint bethoven 2025

rboixaderg opened the pull request:
This pull request introduces a broad set of improvements to ensure compatibility with Volto 18.x.x, update dependencies for modern Python versions, streamline development workflows, and enhance default configuration and content types. The changes affect the CI pipeline, Docker setup, configuration files, default blocks, and key application settings, aligning the project with current standards and Volto requirements.

**Compatibility and Feature Enhancements:**

* Added minimum features for compatibility with Volto 18.x.x and updated the changelog to reflect this milestone.
* Updated application settings in `guillotina_volto/__init__.py` to support new control panels, roles, layouts, workflows, and default blocks; improved integration with Volto and added new content types and blocks (e.g., `Page`, `slate`). [[1]](diffhunk://#diff-26087e0ca029d35aae59c2a4989eec718b57b98d0e3ceca05896b25aceea6594R26-R57) [[2]](diffhunk://#diff-26087e0ca029d35aae59c2a4989eec718b57b98d0e3ceca05896b25aceea6594L64-R84) [[3]](diffhunk://#diff-26087e0ca029d35aae59c2a4989eec718b57b98d0e3ceca05896b25aceea6594R124)
* Registered additional API modules for improved backend coverage and Volto feature support.

**Infrastructure and Dependency Updates:**

* Extended CI and Docker support to Python 3.9, 3.10, and 3.11; upgraded dependencies in `contrib-requirements.txt` and Dockerfile for better compatibility and security. [[1]](diffhunk://#diff-c471496bcb3ed47397b73d2e3e3aa41a8a679c86accaed9e05676fd9d5987434L12-R37) [[2]](diffhunk://#diff-c471496bcb3ed47397b73d2e3e3aa41a8a679c86accaed9e05676fd9d5987434L53-R62) [[3]](diffhunk://#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557L1-R24) [[4]](diffhunk://#diff-231d568eee5eaea2a4e62b5d9dbc1375157b6dd41bfe61a0038e3ca53d98ea5eL4-R14)
* Updated Docker Compose and database configuration to use PostgreSQL 15, new ports, and improved service commands for local development and testing. [[1]](diffhunk://#diff-3fde9d1a396e140fefc7676e1bd237d67b6864552b6f45af1ebcc27bcd0bb6e9L7-R43) [[2]](diffhunk://#diff-f6e99a0e42624f7c35d930ed39895396651afe1295fac0142d0c7a4e365dd539L5-R5) [[3]](diffhunk://#diff-f6e99a0e42624f7c35d930ed39895396651afe1295fac0142d0c7a4e365dd539L22-R22)

**Development Workflow Improvements:**

* Improved Makefile with new targets for starting, stopping, formatting, and running tests, streamlining local development. [[1]](diffhunk://#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L23-R23) [[2]](diffhunk://#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R44-R57)
* Updated documentation with required environment variables for Volto integration.

**Default Content and Block Configuration:**

* Changed default blocks for documents and containers to use the `slate` block instead of `text` or `description`, both in configuration files and Helm charts, ensuring consistency with Volto 18.x.x expectations. [[1]](diffhunk://#diff-d8d0422389f03d783e32e627250fe29834bd09c6361640d1ff00661dd6820034L78-L100) [[2]](diffhunk://#diff-f6e99a0e42624f7c35d930ed39895396651afe1295fac0142d0c7a4e365dd539L79-L101) [[3]](diffhunk://#diff-f1d414b7661a0c814490b6b51a42da5cd5332c7a15a8b5f5ac6b07933e9e5cc0L238-R238) [[4]](diffhunk://#diff-f1d414b7661a0c814490b6b51a42da5cd5332c7a15a8b5f5ac6b07933e9e5cc0L271-R271) [[5]](diffhunk://#diff-f1d414b7661a0c814490b6b51a42da5cd5332c7a15a8b5f5ac6b07933e9e5cc0L282-R291) [[6]](diffhunk://#diff-af3a6a2345623462b227f3f8b70120f8ef422205924fcfb2668b675925d2c098L33-R92)

**Configuration and Indexing Enhancements:**

* Added and updated application and database settings for better extensibility, including new application modules, control panels, and index fields for user objects. [[1]](diffhunk://#diff-26087e0ca029d35aae59c2a4989eec718b57b98d0e3ceca05896b25aceea6594R6-R12) [[2]](diffhunk://#diff-26087e0ca029d35aae59c2a4989eec718b57b98d0e3ceca05896b25aceea6594R152) [[3]](diffhunk://#diff-af3a6a2345623462b227f3f8b70120f8ef422205924fcfb2668b675925d2c098L1-R27)R1, [[4]](diffhunk://#diff-af3a6a2345623462b227f3f8b70120f8ef422205924fcfb2668b675925d2c098R37-R39)

These changes collectively modernize the codebase, improve developer experience, and ensure the project is ready for use with the latest Volto frontend and Python runtimes.

nilbacardit26 committed:
getting connected to volto

nilbacardit26 committed:
adding endpoints and refector testing

nilbacardit26 committed:
creating users without validation

nilbacardit26 committed:
api search users

rboixaderg committed:
refactor: update block types from 'text' to 'slate' in configuration files
feat: add UserIdChecker adapter for username validation

nilbacardit26 committed:
indexing user_name

nilbacardit26 committed:
merge

rboixaderg committed:
feat: add user group settings interface and update README with environment variables; update requirements and enhance history handling

nilbacardit26 committed:
addinh sharing endpoint

rboixaderg committed:
refactor: update role configurations and permissions; enhance sharing roles retrieval; add content rules vocabulary

rboixaderg committed:
chore: update deps, format code and update github actions

rboixaderg committed:
chore: update CI workflow to run pre-checks on guillotina_volto

rboixaderg committed:
chore: remove specific Cython installation from CI workflow

rboixaderg committed:
fix: correct coverage report path in CI workflow for guillotina_volto

rboixaderg committed:
chore: downgrade flake8 version to 7.1.2 in contrib-requirements.txt

rboixaderg committed:
chore: downgrade black version to 24.8.0 in contrib-requirements.txt

rboixaderg committed:
chore: downgrade isort version to 5.10 in contrib-requirements.txt

rboixaderg committed:
chore: downgrade mypy version to 1.14.1 in contrib-requirements.txt

rboixaderg committed:
chore: downgrade pytest-asyncio version to 0.24.0 in requirements-test.txt

rboixaderg committed:
chore: downgrade pytest-cov version to 5.0.0 in requirements-test.txt

rboixaderg committed:
fix: remove redundant reruns option in CI workflow for pytest

rboixaderg committed:
fix: remove unused database password setting in get_db_settings_volto function

rboixaderg committed:
fix: update pytest command in CI workflow to include exit on first failure and add database password for custom DB in fixtures

rboixaderg committed:
fix: update database settings retrieval in fixtures and skip test for non-Postgres environments

rboixaderg committed:
fix: enhance database settings configuration for Postgres in fixtures

rboixaderg committed:
test: skip test for guillotina 7.0.5 due to unimplemented features

nilbacardit26 committed:
adding history fucntionality WIP

nilbacardit26 committed:
adding history functionality

nilbacardit26 committed:
deleting _v_history

nilbacardit26 committed:
adding history get version

rboixaderg committed:
refactor: improve history tracking and event handling in subscribers

- Updated history functions to ensure correct versioning and state tracking.
- Enhanced the handling of object creation and modification events to include detailed history records.
- Added workflow change tracking to the history.
- Adjusted tests to validate new history features and ensure proper assertions on history entries.

nilbacardit26 committed:
renaming CMSFolder to Page

nilbacardit26 committed:
adding expand query in history version

nilbacardit26 committed:
adding language settings in controlpanel

rboixaderg committed:
update: bump guillotina version to 7.0.6 and add python-dateutil dependency; refactor code for consistency and clarity in various files

rboixaderg committed:
feat: enhance permissions and improve history handling

- Added permissions for managing users and accessing the control panel for the SiteAdministrator role.
- Updated history version handling to merge dictionary values instead of overriding them.
- Introduced new tests for history merging and site validation, ensuring proper functionality and user access.

rboixaderg committed:
chore: changelog

nilbacardit26 committed:
WIP languags

nilbacardit26 committed:
functionality adding languages

nilbacardit26 committed:
deleting references language

nilbacardit26 commented on guillotina_volto/content/page.py:
```diff
@@ -1,18 +1,19 @@
 
 
 @configure.contenttype(
-    type_name="CMSFolder",
-    schema=ICMSFolder,
+    type_name="Page",
+    schema=IPage,
```
thanks copilot, I guess no one is using that yet! :)

rboixaderg committed:
refactor: update language handling and event interfaces

- Reintroduced IWorkflowChangedEvent in events.py for better event management.
- Cleaned up imports in controlpanels.py and resource.py for clarity.
- Enhanced language settings in interfaces.py with structured fieldsets for better organization.
- Added a new vocabulary for available languages to support multilingual features.
- Improved subscriber logic for handling language-related events and translations.
- Updated tests to ensure proper functionality of language settings and translations.

rboixaderg committed:
feat: add navigation service for resource access

- Introduced a new Navroot service to provide navigation data for resources.
- Implemented GET method to return the navigation structure, including options to include or omit specific fields.
- Integrated event notification for object visits to enhance tracking and analytics.

rboixaderg committed:
fix: remove event notification from Navroot service

rboixaderg committed:
chore: format

nilbacardit26 committed:
adding translations behavior in language folders

nilbacardit26 committed:
transaltions endpoint fixed

nilbacardit26 committed:
adding language token and title and navroot in default get

nilbacardit26 committed:
adding title and token in language serialization

rboixaderg committed:
feat: add language field if resource has a ILanguageFolder parent

rboixaderg committed:
chore: makefile

rboixaderg committed:
fix: update navigation service to use parent language folder context

- Modified the Navigation service to search for content depth using the parent language folder if available.
- Adjusted the parent UUID reference to align with the new context for navigation items.
- Enhanced language handling in resource responses to utilize the title and ID of the parent language folder.

nilbacardit26 committed:
adding endpoint translator-location

rboixaderg committed:
feat: add querystring endpoints for searching and configuration

- Introduced `@querystring-search` POST endpoint for querying resources with customizable parameters.
- Added `@querystring` GET endpoint to retrieve querystring configuration, including enabled and sortable indexes.
- Implemented vocabulary support for content types and workflow states, enhancing the query capabilities.
- Updated tests to validate the new querystring functionality and vocabulary retrieval.

rboixaderg committed:
feat: add DELETE endpoint for removing translations

- Introduced a new `@translations` DELETE endpoint to allow the removal of specific translations based on language.
- Implemented the `DeleteTranslations` service to handle the deletion logic and notify changes.
- Added tests to validate the unlinking of translations and ensure proper functionality.

nilbacardit26 committed:
adding POST @translatoins and move logic to behavior

nilbacardit26 committed:
format raise exception translation already added

rboixaderg committed:
refactor: streamline event notifications and improve history response

rboixaderg committed:
refactor: update control panel settings and introduce search functionality

- Renamed and updated schemas for image and validation settings in the control panel.
- Removed the obsolete controlpanels.py file.
- Added a new search endpoint to facilitate content searches with customizable parameters.
- Enhanced the IUserGroupSettings interface with additional fields for user and group management.
- Updated tests to reflect changes in history response structure.

nilbacardit26 committed:
disallowing language folder

nilbacardit26 committed:
passing tests

rboixaderg committed:
feat: introduce Page content type and enhance search functionality

- Added a new `Page` content type with associated behaviors and fields for preview images and captions.
- Updated search functionality to include title and description in the response.
- Enhanced querystring capabilities by adding a new `SearchableText` field for improved content searching.
- Refactored existing settings and behaviors to accommodate the new content type and search features.
- Updated tests to reflect changes in content types and search responses.

rboixaderg committed:
chore: update database configurations and enhance local settings

rboixaderg committed:
chore: update .gitignore and refactor language addon settings

nilbacardit26 reviewed (approved):
oh my god :)

We need to refactor some things, but let's merge it
